cni/pkg/cmd/root.go

```diff
@@ -110,17 +114,61 @@ var rootCmd = &cobra.Command{
 			}
 
 			ambientAgent.Start()
-			defer ambientAgent.Stop()
+			// Ambient watch server IS enabled - on shutdown
+			// we need to check and see if this is an upgrade.
+			//
+			// if it is, we do NOT remove the plugin, and do
+			// NOT do ambient watch server cleanup
+			cleanupDefer = func() {
```

Comment:
very nitpick but I would just `defer func()` in two places. Doesn't matter at all, probably not worth commenting but I already typed it :stuck_out_tongue: